Automattic/jetpack
# Upgraded to gulp 4 and moved React and Sass to separate files.

While trying to fix the Beta Builder I refactored the gulpfile thinking that the problem was there. It wasn't, but letting the code go to waste is going to be a shame, so here it is.

#### Changes proposed in this Pull Request:
* Upgrades gulp to version 4.
* Redefines tasks to use series and parallels instead of dependencies, makes for easier management.
* Moves React and SASS tasks to separate files, makes for easier readability.
* Adds a forgotten Search widget CSS file back into the glob.
* Fixes static.html file not being generated in watch mode.
* Fixes static.html file silently failing on errors.

#### Testing instructions:
* Test various `yarn` commands that we generally use for development and deployment.
* See if the admin and client area of Jetpack is fine.
